internal/configuration/flake.go

```diff
@@ -45,7 +74,12 @@ func FlakeRefFromEnv(defaultLocation string) (*FlakeRef, error) {
 		return nil, fmt.Errorf("NIXOS_CONFIG is not set")
 	}
 
-	return FlakeRefFromString(nixosConfig), nil
+	resolvedConfig, err := ResolvePath(nixosConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve path %s: %w", nixosConfig, err)
+	}
+
+	return FlakeRefFromString(resolvedConfig), nil
```

Comment:
This repeats the resolve path logic twice, once in the `FlakeRefFromString()` function, so the original code suffices in this case.